modules/simple-payments/simple-payments.php

```diff
@@ -69,6 +77,33 @@ function get_blog_id() {
 		return Jetpack_Options::get_option( 'id' );
 	}
 
+	/**
+	 * Used to check whether Simple Payments are enabled for given site.
+	 *
+	 * @return bool True if Simple Payments are enabled, false otherwise.
+	 */
+	function is_enabled_jetpack_simple_payments() {
+		/**
+		 * Can be used by plugin authors to disable the conflicting output of Simple Payments.
+		 *
+		 * @since 6.2.0
+		 *
+		 * @param bool True if Simple Payments should be disabled, false otherwise.
+		 */
+		if ( apply_filters( 'jetpack_disable_simple_payments', false ) ) {
+			return false;
+		}
+
+		// For WPCOM sites
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM && function_exists( 'has_blog_sticker' ) ) {
```

Comment:
Do we need to check this by blog sticker? It would be nicer if we check the plan on dotcom the same way we're checking the plan for JP. It would be _waay_ nicer if we could check both the same way but alas... 🙃

Reply:
In wpcom we also have `is_blog_wpcom`, `is_blog_jetpack` available for use. Internally we can tell with a site_id value of 1 or 2

Reply:
I've been looking for a better solution, but using `has_blog_sticker` is a stablished pattern for checking plans in WP.com (premium themes, AT eligibility, all use stickers), and is also following the same patterns already in use to check for WP.com and plan support on Jetpack.
This is mostly because we are doing this with a _Jetpack First_ approach.